string name
---
int32 error_code
